### PR TITLE
Accept multiple data formats via drag-and-drop in wxDataViewCtrl

### PR DESCRIPTION
**Pull request by valid-ptr**
Allow specifying multiple formats to be accepted when dragging data to
wxDataViewCtrl in the generic and Cocoa implementations.

Add wxDataViewCtrlBase::EnableDropTarget() overload taking an array of
wxDataFormats to support this at the API level.

Add new DoEnableDropTarget() used by both EnableDropTarget() overloads
and implement it in the generic and Cocoa ports. GTK implementation
still uses only a single format, as before.

Also refactor the Cocoa implementation: all operations using dragged
data are now handled by wxDropTarget and unnecessary DataViewPboardType
as removed.

Update the dataview sample to show the new functionality.

See PR: #2467

### CODE REVIEW

**Comment by vadz** [comment]
Oh, and this also omit all my commits from #2467, e.g. the one replacing the array with a vector that I think is really worth having as well as a few tweaks to the docs. Did you omit them intentionally because you don't like these changes or was it just an accident?

**Comment by valid-ptr** [comment]
I've merged your commits!
Everything looks good and works for me!

File: src/osx/cocoa/dataview.mm [hunk 1]
```diff
@@ -1653,8 +1499,6 @@ -(id) init
         currentlyEditedColumn =
             currentlyEditedRow = -1;
 
-        [self setAutoresizesOutlineColumn:NO];
```
Comment (vadz) [inline src/osx/cocoa/dataview.mm:1656]: I think this line was removed accidentally, so I'm going to restore it, please let me know if I'm wrong.

File: src/osx/cocoa/dataview.mm [hunk 2]
```diff
@@ -1653,8 +1499,6 @@ -(id) init
...
-        [self registerForDraggedTypes:[NSArray arrayWithObjects:DataViewPboardType,NSStringPboardType,nil]];
```
Comment (vadz) [inline src/osx/cocoa/dataview.mm:1657]: I _guess_ this line was removed intentionally, so I'm not changing anything here, but please double check it to confirm.

File: src/osx/cocoa/dataview.mm [hunk 3]
```diff
@@ -2500,9 +2344,20 @@ void UpdateWithRow(int row)
...
 {
-    [m_OutlineView reloadItem:[m_DataSource getDataViewItemFromBuffer:item]];
+    if (GetSortingColumn())
+    {
+        if (parent.IsOk())
+            [m_OutlineView reloadItem:[m_DataSource getDataViewItemFromBuffer:parent] reloadChildren:YES];
+        else
+            [m_OutlineView reloadData];
+    }
+    else
```
Comment (vadz) [inline src/osx/cocoa/dataview.mm:2356]: This change seems unrelated, so I'm going to undo it, please let me know if I'm missing something here and it's really needed.

**Review by vadz — COMMENTED** [review]
Thanks for these changes and sorry for the delay with getting back to them!

It's great to see how much less code is now needed under Mac, I have no idea why was the old version so much longer and more complicated, but the new one clearly looks much better. However I think that you may have resolved a merge incorrectly because some of the changes to Mac code seem unrelated and don't make sense to me, so I've reverted them.

I've also made `wxDataViewEvent::GetDataBuffer()` work again because we can't silently break the existing code like this (this method would start returning null pointer after your changes, resulting in the existing code using it not working at best and crashing at worst). Using a buffer to copy the data is not ideal and could be avoided by reusing the existing data objects, i.e. returning pointer to their existing buffer rather than a copy, but I didn't have time to do this -- if you can optimize this to avoid this extra copy, it would be welcome.

But this could be done later, for now I think we should merge #2616. Please don't hesitate to review my changes there (especially the ones reverting some of your changes). TIA!

**Comment by vadz** [comment]
Replaced by #2616.
